## Einstein's Mission

Einstein Bannerjee is the name of a fictional character.

Use artificial intelligence together with humanity's accumulated astronomical knowledge to discover, understand, and document interesting objects in the Universe.

# Notebook 017 — Building Einstein's First Scientific Object Profile

## Transforming Astronomical Database Queries into Scientific Knowledge

---

# Einstein's Mission

> **Use artificial intelligence together with humanity's accumulated
> astronomical knowledge to discover, understand, and document
> interesting objects in the Universe.**

Einstein is an AI-augmented astronomical research assistant developed as
part of the **Cosmic Intelligence Lab** project.

Its purpose is not to replace scientific reasoning, but to assist it by
combining image analysis, machine learning, astronomical databases, and
human curiosity.

Every notebook expands Einstein's capabilities while documenting our
shared journey of learning in astronomy and artificial intelligence.

---

# Research Question

After identifying an astronomical object in SIMBAD, can Einstein
construct a comprehensive scientific profile that combines its own
measurements with the accumulated knowledge of the astronomical
community?

Rather than viewing an object simply as a collection of pixels,
Einstein begins to understand it as a physical object within the
Universe.

---

# Objective

In Notebook 016 Einstein learned how to identify an AI-discovered object
using its celestial coordinates.

In this notebook Einstein takes the next step by organizing published
astronomical information into a structured scientific profile.

The goals are to:

1. Retrieve the published identification of an object.
2. Record its astronomical classification.
3. Collect basic observational information.
4. Compare published information with Einstein's own measurements.
5. Create Einstein's first Scientific Object Profile.

---

# Scientific Motivation

Modern astronomy is built upon generations of observations, catalogues,
and published research.

Artificial intelligence can identify interesting objects within an
image, but understanding those objects requires connecting image-derived
measurements with humanity's accumulated scientific knowledge.

This notebook represents Einstein's first attempt to organize that
knowledge into a form suitable for scientific investigation and future
research.

---

# Einstein's Research Workflow

```
AI Discovery
      ↓
Sky Coordinates
      ↓
SIMBAD Identification
      ↓
Published Information
      ↓
Scientific Object Profile
      ↓
Research Journal
```

In [1]:
# Cell 1a - Investigation target

TARGET_LABEL = 45

In [2]:
# ============================================================
# Cell 3 - Load Einstein Master Catalog
# ============================================================

from pathlib import Path
from astropy.table import Table

catalog_file = Path("../data/catalogs/einstein_master_catalog.ecsv")

catalog = Table.read(catalog_file)

print("=" * 60)
print("Einstein Master Catalog")
print("=" * 60)

print(f"Objects loaded : {len(catalog)}")

Einstein Master Catalog
Objects loaded : 79


In [3]:
# ============================================================
# Cell 4 - Select Einstein's First Target
# ============================================================

target = catalog[catalog["label"] == TARGET_LABEL][0]

print("=" * 60)
print("Einstein's First Target")
print("=" * 60)

print(f"Label : {target['label']}")
print(f"Flux  : {target['segment_flux']:.2f}")
print(f"RA    : {target['ra_deg']:.6f}°")
print(f"Dec   : {target['dec_deg']:.6f}°")

Einstein's First Target
Label : 45
Flux  : 137.40
RA    : 53.169936°
Dec   : -27.771127°


In [4]:
# ============================================================
# Cell 5 - Configure SIMBAD
# ============================================================

from astroquery.simbad import Simbad

custom_simbad = Simbad()

custom_simbad.add_votable_fields(
    "otype",
    "ra",
    "dec"
)

print("SIMBAD configured.")

SIMBAD configured.


In [5]:
# ============================================================
# Cell 6 - Einstein's First SIMBAD Query
# ============================================================

from astropy.coordinates import SkyCoord
import astropy.units as u

# Build SkyCoord object
coord = SkyCoord(
    ra=target["ra_deg"] * u.deg,
    dec=target["dec_deg"] * u.deg,
    frame="icrs"
)

# Search within 5 arcseconds
result = custom_simbad.query_region(
    coord,
    radius=5 * u.arcsec
)

print("=" * 60)
print("SIMBAD Query Result")
print("=" * 60)

if result is None:
    print("No object found.")
else:
    print(result)

SIMBAD Query Result
        main_id                 ra        ...     coo_bibcode     otype
                               deg        ...                          
----------------------- ----------------- ... ------------------- -----
        [RTG2015] 24643         53.171235 ... 2015AJ....150...31R     G
         [RTG2015] 8918         53.170099 ... 2015AJ....150...31R     G
         [RTG2015] 8953         53.171081 ... 2015AJ....150...31R     G
        [RTG2015] 24629         53.168661 ... 2015AJ....150...31R     G
        [RTG2015] 24544         53.169933 ... 2015AJ....150...31R     G
[SWM2014] GOODS-S 30321 53.16975000000001 ... 2018ApJS..237...12O     G
         [RTG2015] 9115         53.168795 ... 2015AJ....150...31R     G
        [RTG2015] 24559         53.170174 ... 2015AJ....150...31R     G
              UDF  7858        53.1712265 ... 2005AJ....130....1T     G
                    ...               ... ...                 ...   ...
       [WDA2016] 3 mm.8           53.1695 ..

In [6]:
# ============================================================
# Inspect the SIMBAD result table
# ============================================================

print(result)

print("\nColumn names:")
print(result.colnames)

        main_id                 ra        ...     coo_bibcode     otype
                               deg        ...                          
----------------------- ----------------- ... ------------------- -----
        [RTG2015] 24643         53.171235 ... 2015AJ....150...31R     G
         [RTG2015] 8918         53.170099 ... 2015AJ....150...31R     G
         [RTG2015] 8953         53.171081 ... 2015AJ....150...31R     G
        [RTG2015] 24629         53.168661 ... 2015AJ....150...31R     G
        [RTG2015] 24544         53.169933 ... 2015AJ....150...31R     G
[SWM2014] GOODS-S 30321 53.16975000000001 ... 2018ApJS..237...12O     G
         [RTG2015] 9115         53.168795 ... 2015AJ....150...31R     G
        [RTG2015] 24559         53.170174 ... 2015AJ....150...31R     G
              UDF  7858        53.1712265 ... 2005AJ....130....1T     G
                    ...               ... ...                 ...   ...
       [WDA2016] 3 mm.8           53.1695 ... 2016ApJ...833...67

In [7]:
# ============================================================
# Cell 7 - Einstein Interprets the Result
# ============================================================

if result is None:
    print("No SIMBAD object found.")

else:

    obj = result[0]

    print("=" * 60)
    print("Einstein's First Identification")
    print("=" * 60)

    print(f"Object Name : {obj['main_id']}")
    print(f"Object Type : {obj['otype']}")
    print(f"RA          : {obj['ra']}")
    print(f"Dec         : {obj['dec']}")

Einstein's First Identification
Object Name : [RTG2015] 24643
Object Type : G
RA          : 53.171235
Dec         : -27.771108


# Einstein's First Scientific Identification

Property	Value
Einstein Label	45
SIMBAD Identifier	[RTG2015] 24643
Object Type	G -----------> (galaxy)
Right Ascension	53.171235°
Declination	−27.771108°

## First Scientific Identification

Einstein has successfully identified its first AI-discovered object
using the SIMBAD astronomical database.

Object **Label 45**, originally detected in the Hubble image and
classified through machine learning, has been matched with the
published galaxy **[RTG2015] 24643**.

This represents the first successful integration of Einstein's own
measurements with humanity's accumulated astronomical knowledge.

From this point forward, Einstein is no longer limited to measuring
objects—it can begin learning what astronomers already know about them.

In [8]:
# ============================================================
# Cell 8 - Einstein's First Scientific Object Profile
# ============================================================

profile = {
    "Einstein Label": int(target["label"]),
    "SIMBAD Name": str(obj["main_id"]),
    "Object Type": str(obj["otype"]),

    # Einstein measurements
    "Measured Flux": float(target["segment_flux"]),
    "Area (pixels)": int(target["area"]),
    "Semimajor Axis (pixels)": float(target["semimajor_axis"]),
    "Semiminor Axis (pixels)": float(target["semiminor_axis"]),
    "Orientation (deg)": float(target["orientation"]),
    "Eccentricity": float(target["eccentricity"]),

    # Coordinates
    "Einstein RA (deg)": float(target["ra_deg"]),
    "Einstein Dec (deg)": float(target["dec_deg"]),
    "SIMBAD RA": str(obj["ra"]),
    "SIMBAD Dec": str(obj["dec"]),
}

print("=" * 65)
print("Einstein Scientific Object Profile")
print("=" * 65)

for key, value in profile.items():
    print(f"{key:<28} : {value}")

Einstein Scientific Object Profile
Einstein Label               : 45
SIMBAD Name                  : [RTG2015] 24643
Object Type                  : G
Measured Flux                : 137.40018010139465
Area (pixels)                : 142
Semimajor Axis (pixels)      : 3.615779764564503
Semiminor Axis (pixels)      : 2.411772662245055
Orientation (deg)            : 87.63921016043626
Eccentricity                 : 0.7450459404507496
Einstein RA (deg)            : 53.16993598680541
Einstein Dec (deg)           : -27.771127189673585
SIMBAD RA                    : 53.171235
SIMBAD Dec                   : -27.771108


In [9]:
# ============================================================
# Cell 9 - Einstein's Interpretation
# ============================================================

print("=" * 60)
print("Einstein's Scientific Interpretation")
print("=" * 60)

print(f"""
Object {profile['Einstein Label']} has been successfully matched
with the SIMBAD database.

Published Identifier:
    {profile['SIMBAD Name']}

Classification:
    {profile['Object Type']} (Galaxy)

Einstein's measurements indicate an extended object with:

    Flux              : {profile['Measured Flux']:.2f}
    Area              : {profile['Area (pixels)']} pixels
    Semimajor Axis    : {profile['Semimajor Axis (pixels)']:.2f}
    Semiminor Axis    : {profile['Semiminor Axis (pixels)']:.2f}
    Orientation       : {profile['Orientation (deg)']:.2f}°
    Eccentricity      : {profile['Eccentricity']:.3f}

Conclusion

Einstein has successfully combined image-derived measurements with information 
retrieved from the SIMBAD astronomical database to produce its first Scientific 
Object Profile.

The identification of Object 45 as the galaxy [RTG2015] 24643 demonstrates that 
AI-generated detections can be linked directly to the astronomical literature, 
providing a foundation for richer scientific investigations in future notebooks.

This marks Einstein's first successful Scientific Object Profile.
""")

Einstein's Scientific Interpretation

Object 45 has been successfully matched
with the SIMBAD database.

Published Identifier:
    [RTG2015] 24643

Classification:
    G (Galaxy)

Einstein's measurements indicate an extended object with:

    Flux              : 137.40
    Area              : 142 pixels
    Semimajor Axis    : 3.62
    Semiminor Axis    : 2.41
    Orientation       : 87.64°
    Eccentricity      : 0.745

Conclusion

Einstein has successfully combined image-derived measurements with information 
retrieved from the SIMBAD astronomical database to produce its first Scientific 
Object Profile.

The identification of Object 45 as the galaxy [RTG2015] 24643 demonstrates that 
AI-generated detections can be linked directly to the astronomical literature, 
providing a foundation for richer scientific investigations in future notebooks.

This marks Einstein's first successful Scientific Object Profile.



In [10]:
# ============================================================
# Cell 10 - Save Scientific Object Profile
# ============================================================

from astropy.table import Table
from pathlib import Path

profile_table = Table(rows=[profile])

output_dir = Path("../data/catalogs")
output_dir.mkdir(parents=True, exist_ok=True)

outfile = output_dir / "object45_profile.ecsv"

profile_table.write(outfile, format="ascii.ecsv", overwrite=True)

print(f"Scientific profile saved to:\n{outfile}")

Scientific profile saved to:
../data/catalogs/object45_profile.ecsv


In [11]:
from astropy.coordinates import SkyCoord
import astropy.units as u

einstein_coord = SkyCoord(
    ra=target["ra_deg"] * u.deg,
    dec=target["dec_deg"] * u.deg
)

simbad_coord = SkyCoord(
    ra=float(obj["ra"]) * u.deg,
    dec=float(obj["dec"]) * u.deg
)

sep = einstein_coord.separation(simbad_coord)

print(f"Coordinate separation: {sep.arcsec:.2f} arcsec")

Coordinate separation: 4.14 arcsec


In [12]:
# ============================================================
# Notebook Status
# ============================================================

from datetime import datetime

print("=" * 60)
print("Notebook Status")
print("=" * 60)

print("Status    : PASS")
print("Notebook  : 17_scientific_object_profiles.ipynb")
print("Completed :", datetime.now().strftime("%Y-%m-%d %H:%M"))

Notebook Status
Status    : PASS
Notebook  : 17_scientific_object_profiles.ipynb
Completed : 2026-07-19 16:15
